RELEASE.md

```diff
@@ -0,0 +1,39 @@
+Release type: minor
+
+This release improves how we deal with custom scalars. Instead of being global
+they are now scoped to the schema. This allows you to have multiple schemas in
+the same project with differnet scalars.
+
+Also you can now override the built in scalars with your own custom
+implementation. Out of the box Strawberry provides you with custom scalars for
+common Python types like `datetime` and `Decimal`. If you require a custom
+implementation of one of these built in scalars you can how subclass the
+`Schema` class and override the `get_scalar` function to return your own custom
+implementation:
+
+```python
+from datetime import datetime, timezone
+import strawberry
+
+EpocDateTime = strawberry.scalar(
+    datetime,
+    serialize=lambda value: int(value.timestamp()),
+    parse_value=lambda value: datetime.fromtimestamp(int(value), timezone.utc),
+)
+
+class MySchema(strawberry.Schema):
+    def get_scalar(self, scalar):
+        if scalar == datetime:
+            return EpocDateTime
+        return super().get_scalar(scalar)
```

Comment:
This seems a bit redundant as `datetime` is the first argument to `strawberry.scalar`. I wonder if there's a different mechanism we could use to have a schema "use" a scalar?

Reply:
I don't follow. We're not creating the scalar here, we're just processing the scalars that are already defined in the schema.

Reply:
I think my concern was that we first define the custom scalar by saying that it works _for_ `datetime`:
```python
EpochDateTime = strawberry.scalar(
    datetime,
    ...,
)
```
and then when we want to _use_ the scalar with a schema, we again say that the scalar is being used _for_ `datetime`:
```python
schema = strawberry.Schema(
  Query,
  scalar_overrides={
    datetime: EpochDateTime,
  }
)
```

Can the `scalar_overrides` just take a list of Scalars and figure out what they should be applied to by looking inside the custom scalar?
 ```python
schema = strawberry.Schema(
  Query,
  scalar_overrides=[EpochDateTime],
)
```

I guess the disadvantage of this is that you can't, for example, make a custom scalar that works for all numbers (i.e. `float`), and then tell the schema to use it only for `int`s?

Reply:
Oh I see. The issue is that you don't have to provide a first param to `strawberry.scalar`, it's mainly a helper function: https://github.com/strawberry-graphql/strawberry/blob/71e6adb743df4d2830b5a3b046d33adf40daddc3/strawberry/custom_scalar.py#L90-L96

So I'm not sure we could reliably replace the correct scalar types.

Reply:
strawberry.scalar should return the class, no? so this should be doable I think 😊

Reply:
Hmm possibly. I think the mapping is clearer though

Reply:
yes, I think to too :)

I think we can add some error checking in another PR, for example in this case:

```
schema = strawberry.Schema(
  Query,
  scalar_overrides={
    int: EpochDateTime,
  }
)
```